lib/class-wp-rest-menus-controller.php

```diff
@@ -431,6 +480,62 @@ public function delete_item( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Returns the value of a menu's auto_add
+	 *
+	 * @param int $menu_id The menu id to update the location form.
+	 *
+	 * @return Boolean The value of auto_add.
+	 */
+	function get_menu_auto_add( $menu_id ) {
+		$nav_menu_option = (array) get_option( 'nav_menu_options', array( 'auto_add' => array() ) );
+		if ( ! in_array( $menu_id, $nav_menu_option['auto_add'], true ) ) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * Updates the menu's auto add from a REST request.
+	 *
+	 * @param int             $menu_id The menu id to update the location form.
+	 * @param WP_REST_Request $request The request object with menu and locations data.
+	 *
+	 * @return Boolean|Null   Return the value of auto_add or null if the auto_add
+	 *                        is not present in the request.
+	 */
+	function handle_auto_add( $menu_id, $request ) {
+		if ( ! isset( $request['auto_add'] ) ) {
+			return true;
+		}
+
+		$nav_menu_option = (array) get_option( 'nav_menu_options', array( 'auto_add' => array() ) );
+
+		if ( ! isset( $nav_menu_option['auto_add'] ) ) {
+			$nav_menu_option['auto_add'] = array();
+		}
+
+		$auto_add = (bool) $request['auto_add'];
+
+		if ( $auto_add ) {
+			if ( ! in_array( $menu_id, $nav_menu_option['auto_add'], true ) ) {
+				$nav_menu_option['auto_add'][] = $menu_id;
+			}
+		} else {
+			$key = array_search( $menu_id, $nav_menu_option['auto_add'], true );
+			if ( false !== $key ) {
+				unset( $nav_menu_option['auto_add'][ $key ] );
```

Comment:
Should we reorder this with `array_values()` so the indexes are consecutive?

Reply:
I think you are right, we should do something like this -

https://github.com/WordPress/wordpress-develop/blob/d8c502e9663cf1c266aed375d4ba2e09a86a4f2d/src/wp-includes/customize/class-wp-customize-nav-menu-setting.php#L600-L615

This is elsewhere in core, so I may just copy and paste it over